Infrastructure/CosmosDbService.cs

```diff
@@ -0,0 +1,72 @@
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using Database.Models;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Cosmos.Fluent;
+
+namespace incrementally.Services
+{
+    public class CosmosDbService : IDatabaseConnector
+    {
+        private Dictionary<string, Container> _containers;
+        private CosmosClient _client;
+        private DatabaseResponse _database;
+        private string _databaseName;
+
+        public CosmosDbService(string databaseName)
+        {
+            _databaseName = databaseName;
+            _containers = new Dictionary<string, Container>();
+        }
+
+        public async Task Initialize(List<string> containerNames, string account, string key)
+        {
+            CosmosClientBuilder clientBuilder = new CosmosClientBuilder(account, key);
+            _client = clientBuilder
+              .WithConnectionModeDirect()
+              .Build();
+            _database = await _client.CreateDatabaseIfNotExistsAsync(_databaseName);
+            containerNames.ForEach(async containerName => await CreateContainer(containerName));
+        }
+
+        public async Task CreateContainer(string containerName)
+        {
+            _containers.Add(containerName, _client.GetContainer(_databaseName, containerName));
+            await _database.Database.CreateContainerIfNotExistsAsync(containerName, "/id");
+        }
+
+        public async Task DeleteRecording(string id)
+        {
+            await _containers["recordings"].DeleteItemAsync<RecordingMetadata>(id, new PartitionKey(id));
+        }
+
+        public async Task AddRecording(RecordingMetadata recordingMetadata)
+        {
+            await _containers["recordings"].CreateItemAsync(recordingMetadata, new PartitionKey(recordingMetadata.Id));
+        }
+
+        public async Task<IEnumerable<RecordingMetadata>> GetTopRecordings()
+        {
+            var query = new QueryDefinition("SELECT * FROM c ORDER BY c.createdAt DESC OFFSET 0 LIMIT 10");
```

Comment:
For simple queries like these I don't mind the SQL, but In my experience there is often a lot to be gained by making use of some repository-layer tools. In Java this would be JPA/Hibernate (sorry to be quoting java all the time), and from a quick search the .Net alternative seems to be .Net Persistance Api. Again I'm not experienced with this, but thats my immediate reaction.

Reply:
You are right, SQL directly in the code like this is probably not the way to go. I think I will create a separate task for this. The tool you're looking for is [EF Core](https://docs.microsoft.com/en-us/ef/core/) ;)